tools-local/unix/tool-helper.sh

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env bash
+
+# Provides helper functions.
+
+# Gets name of the operating system.
+# Exit 1 if unable to get operating system name.
+get_os_name()
+{
+    osName="$(uname -s)"
+
+    if [ $? -ne 0 ] || [ -z "$osName" ]; then
+        echo "Unable to determine the name of the operating system."
+        exit 1
+    fi
+
+    if echo "$osName" | grep -iqF "Darwin"; then
+        osName="OSX"
+    else
+        osName="Linux"
+    fi
+
+    echo "$osName"
+}
+
+# Eval .toolversions file.
+# TODO: 
```

Comment:
Agree.  Three issues to log are as follows:

1. Consider accepting the path to an override .toolversions file. If the override .toolversions is available then, use the config values from that file. If override is not available then use the default .toolversions file.
2. Supporting spaces in paths is something we should do before merging this to CoreFX to avoid adding debt, but we shouldn't fix it yet so that we focus on the core concepts in this PR.
3. It might be ok to set a prerequisite for running a script that the helper has already been dot-sourced. This avoids the inefficiency and code duplication of dot-sourcing it within every script, but it makes it harder to run each script individually without this knowledge.